homeassistant/components/broadlink/select.py

```diff
@@ -0,0 +1,78 @@
+"""Support for Broadlink selects."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BroadlinkDevice
+from .const import DOMAIN
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink select."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+    async_add_entities([BroadlinkDayOfWeek(device)])
+
+
+class BroadlinkDayOfWeek(BroadlinkEntity, SelectEntity):
+    """Representation of a Broadlink day of week."""
+
+    _attr_has_entity_name = True
+    _attr_current_option: str | None = None
+    _attr_options = [
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "6",
+        "7",
+    ]
+    _attr_translation_key = "day_of_week"
+
+    def __init__(self, device: BroadlinkDevice) -> None:
+        """Initialize the select."""
+        super().__init__(device)
+
+        self._attr_unique_id = f"{device.unique_id}-dayofweek"
+
+    def _update_state(self, data: dict[str, Any]) -> None:
+        """Update the state of the entity."""
+        if data is None or "dayofweek" not in data:
+            self._attr_current_option = None
+        else:
+            self._attr_current_option = str(data["dayofweek"])
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected option."""
+        if (
+            self._coordinator.data is None
+            or "hour" not in self._coordinator.data
+            or "min" not in self._coordinator.data
+            or "sec" not in self._coordinator.data
+        ):
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="request_failed_device_not_connected",
+            )
```

Comment:
Should we make the entity unavailable when this happens?

Reply:
Sorry. Was a leftover.